projects/graph/src/graph.py

```diff
@@ -1,9 +1,45 @@
-"""
-Simple graph implementation compatible with BokehGraph class.
-"""
+class Vertex:
+    def __init__(self, n):
+        self.name = n
+        self.neighbors = list()
 
+    def add_neighbor(self, vertex):
+        if vertex not in self.neighbors:
+            self.neighbors.append(v)
+            self.neighbors.sort()
 
 class Graph:
-    """Represent a graph as a dictionary of vertices mapping labels to edges."""
-    def __init__(self):
-        pass  # TODO
+    vertices = {}
+
+    def add_vertex(self, vertex):
+        if isinstance(vertex, Vertex) and vertex.name not in self.vertices:
+            self.vertices[vertex.name] = vertex
+            return True
+        else:
+            return False
+
+    def add_edge(self, v1, v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            for key, value in self.vertices.items():
```

Comment:
your vertices is a dictionary (JS object). so if you already know the key, you can do something like `self.vertices[v1].neighbors.append(v2)`